Dockerfile

```diff
@@ -1,21 +1,26 @@
 # syntax=docker/dockerfile:experimental
 
 # Build the controller binary
-FROM --platform=${TARGETPLATFORM} golang:1.17 as builder
+FROM --platform=${BUILDPLATFORM} golang:1.17 as builder
 
 WORKDIR /workspace
 
-COPY go.mod go.sum ./
+ARG GOPROXY
+ENV GOPROXY=${GOPROXY}
 
 # uncomment if using vendor
 # COPY ./vendor ./vendor
 
-ARG GOPROXY
-ENV GOPROXY=${GOPROXY}
-
+COPY go.mod go.sum ./
 RUN go mod download
 
-COPY . ./
+COPY ./main.go ./ATTRIBUTION.txt ./
+COPY .git/ ./.git/
+COPY pkg/ ./pkg/
+COPY apis/ ./apis/
+COPY controllers/ ./controllers/
+COPY mocks/ ./mocks/
+COPY webhooks/ ./webhooks/
```

Comment:
This wasn't strictly related, but I tweaked this dockerfile to be mildly more performant under some circumstances. Particularly: it uses the host build architecture for the building (which leads to fewer total actions in cross-building scenarios), and narrowed the scope of copied files to avoid spurious rebuilds.